converter/service/converter.py

```diff
@@ -0,0 +1,175 @@
+            return
+
+        user_agent = s.headers.get(header_user_agent)
+        accept = s.headers.get(header_accept)
+        decoded_url = query.get(param_url)
+
+        try:
+            json_response = handle(decoded_url, user_agent, accept)
+            s.send_response(HTTPStatus.OK)
+            content_header = gen_universe.format_universe_repo_content_type(
+                _get_repo_version(accept))
+            s.send_header(header_content_type, content_header)
+            s.end_headers()
+            s.wfile.write(json_response.encode())
+        except Exception as e:
+            s.send_error(HTTPStatus.BAD_REQUEST, str(e))
+
+
+def handle(decoded_url, user_agent, accept):
+    """Results in either an error or the requested json data
+
+    :param decoded_url: The url to be fetched from
+    :type decoded_url: str
+    :param user_agent: User-Agent header value
+    :type user_agent: str
+    :param accept: Accept header value
+    :return Requested json data
+    :rtype str (a valid json object)
+    """
+    repo_version = _get_repo_version(accept)
+    dcos_version = _get_dcos_version(user_agent)
+    logger.debug('Url %s\nAgent %s\nAccept %s\nVersion %s\nDC/OS %s',
+                 decoded_url, user_agent, accept, repo_version, dcos_version)
+
+    req = Request(decoded_url)
+    req.add_header(header_user_agent, user_agent)
+    req.add_header(header_accept, accept)
+    res = urlopen(req)
+    charset = res.info().get_param(param_charset) or default_charset
+    packages = json.loads(res.read().decode(charset)).get(json_key_packages)
+    return render_json(packages, dcos_version, repo_version)
+
+
+def render_json(packages, dcos_version, repo_version):
+    """Returns the json
+
+    :param packages: package dictionary
+    :type packages: dict
+    :param dcos_version: version of dcos
+    :type dcos_version: str
+    :param repo_version: version of universe repo
+    :type repo_version: str
+    :return filtered json data based on parameters
+    :rtype str
+    """
+    processed_packages = gen_universe.filter_and_downgrade_packages_by_version(
+        packages,
+        dcos_version
+    )
+    updated_json_data = json.dumps({json_key_packages: processed_packages})
+    errors = gen_universe.validate_repo_with_schema(
+        json.loads(updated_json_data),
+        repo_version
+    )
+    if len(errors) != 0:
+        logger.error(errors)
+        raise ValueError(ErrorResponse.VALIDATION_ERROR)
+    return updated_json_data
+
+
+def _get_repo_version(accept_header):
+    """Returns the version of the universe repo parsed.
+    # TODO Should support versions of two digits - y/n ?
+    :param accept_header: String
+    :return repo version as str or Error
+    :rtype str
+    """
+    result = re.findall(r'\bversion=v\d{1,2}', accept_header)
+    if result is None or len(result) is 0:
+        raise ValueError(ErrorResponse.UNABLE_PARSE)
```

Comment:
This and the error message at L160 should be different so that the caller can tell which header is the problem